osu.Game/Users/UserStatistics.cs

```diff
@@ -26,17 +26,22 @@ public struct LevelInfo
             public int Progress;
         }
 
-        [JsonProperty(@"pp")]
-        public decimal? PP;
+        [JsonProperty(@"rank")]
+        public UserRanks Ranks;
 
-        [JsonProperty(@"pp_rank")] // the API sometimes only returns this value in condensed user responses
-        private int? rank
+        // eventually UserRanks object will be completely replaced with separate global and country rank properties, see https://github.com/ppy/osu-web/blob/cb79bb72186c8f1a25f6a6f5ef315123decb4231/app/Transformers/UserStatisticsTransformer.php#L53.
+        // but for now, always point UserRanks.Global to the global_rank property, as that is included solely for requests like GetUsersRequest.
+        [JsonProperty(@"global_rank")]
+        private int? globalRank
         {
             set => Ranks.Global = value;
```

Comment:
I'm not sure I understand how this works. You have `Ranks` above being populated from the incoming request and then overwritten by this? Is some kind of documentation here missing?

Reply:
It's done in a way that if a `global_rank` property exists, then let it overwrite `Ranks.Global`, because, on cases like old saved tournament bracket json files, the global rank would still be stored at the JSON property `rank.global` instead of `global_rank`.

Reply:
Is that the only case? Sounds like something that should be handled locally if so?

Reply:
I think so, not sure how to handle it locally though.

Reply:
Making it perform the fetch again if null is the easiest method. Assuming that's the only usage.

Reply:
Alright, I've done it locally as you said. Shall I nuke `Ranks.Global` and let all usages use a `GlobalRank` field instead? and if so, shall I do the same with `Ranks.Country` and remove the entire `UserRanks` object?

I was initially going to do that but felt like it may fill the PR with somewhat irrelevant file changes.

Reply:
Is this change not pushed?

Reply:
hmm, due to the issues found in https://github.com/ppy/osu/pull/11799, the user IDs may likely not be set to the correct ones, causing the re-fetching to not work as intended. 

I believe the only way would be to either leave this as-is or make a custom JSON converter or otherwise to migrate `ranks.global` to `global_rank` in tournament client. What do you think?

Reply:
The failures you found are likely not an issue in any normal usage of the system. I would ignore it.